### PR TITLE
Centralized handling of mod channel check

### LINKED ISSUE #1175: Modmail threads don't show total infractions
Related to issues:
https://github.com/python-discord/bot/issues/1127
https://github.com/python-discord/bot/issues/1125

In Modmail threads, notes are listed as necessary, but when issuing an infraction the total number of infractions isn't shown.

### PR DESCRIPTION
**Pull request by mbaruh**
Closes #1175 

The purpose of this PR is to provide one location in the code which is responsible for checking whether a given channel is a mod channel.

Currently this check is performed in the following locations:
* `!user` - If it's a mod channel, it will also display notes and nominations.
* The moderation management cog - only works in mod channels.
* When issuing infractions - mod channels will display infraction number, and total number of infractions for the user.

Issues begin when the definition of what's considered a mod channel changes. The most recent occurrence is the introduction of Modmail, which led to issue https://github.com/python-discord/bot/issues/1125.
However, because those checks are scattered, after that issue was resolved we were still left with https://github.com/python-discord/bot/issues/1175 - which is closed by this PR.

On top of that, despite `#mod-alerts` being whitelisted as a mod channel for quite some time, it didn't display infractions total as that is a separate check, leading to https://github.com/python-discord/bot/issues/1127.

Additionally, I changed all log channels to be considered mod channels, as there were instances where we wanted to run an infraction search in one of the log channels that wasn't `#mod-alerts` but couldn't.

In the future, if we wish to change the definition of what a mod channel is, we can just change the function `is_mod_channel` I'm adding here, and all related commands will work accordingly.

### CODE REVIEW

File: bot/utils/channel.py
Comment (kwzrd) [inline bot/utils/channel.py:25]: Prevailing style for a multi-line expression would be:
```python
return (
    a
    or b
)
```

How would you feel about handling each case separately and logging the decision made? For example:
```python
def is_mod_channel(channel: discord.TextChannel) -> bool:
    """True if `channel` is considered a mod channel."""
    if channel.id in constants.MODERATION_CHANNELS:
        log.trace(f"Channel #{channel} is a configured mod channel")
        return True

    mod_categories = Categories.modmail, Categories.logs
    if any(is_in_category(channel, category) for category in mod_categories):
        log.trace(f"Channel #{channel} is in a configured mod category")
        return True

    log.trace(f"Channel #{channel} is not a mod channel")
    return False
```

Looking at one of the callers, the infraction scheduler, it branches based on the return value just so that it can log the decision that was made. One of the clauses could be removed if we just log here. But it's just a thought, I'm not demanding this.

Finally, I wonder if the mod categories should be defined in the config like the mod channels, and then accessed here as e.g. `constants.MODERATION_CATEGORIES`. I don't necessarily think that's the best way to do it, but it'd be more consistent with the way the channel IDs are handled.

Comment (mbaruh) [reply bot/utils/channel.py:25]: Ok, I'll fix the expression.
Yeah, I think adding the constant is a good idea, it will indicate why those categories are needed in the config file.

As for logging, I think it largely depends on the level of granularity the core developers want for it. Handling it outside of the function allows you to add relevant context. I'm fine with whatever you think is better.

File: bot/exts/moderation/infraction/management.py
Comment (kwzrd) [inline bot/exts/moderation/infraction/management.py:300]: Not part of this PR's diff, but I feel like we could make the first predicate simpler. Checking the intersection of `ctx.author`'s role ids and the constants should work, unless I'm missing something.

**Review by kwzrd — APPROVED** [review]
Thanks, I think it's better this way. Logging the exact decision helps a lot during testing. The caller can always log more context ~ both before, and after the check ~ if necessary.

The code is functional and I agree that this is an improvement. Please double-check the configured category IDs. I cannot help verify them as I don't see those channels.

It's good to include the phrase "Closes #1175" ([or a variation thereof](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords#about-issue-references)) in the PR description, it will auto-close the issue once we merge this PR.

File: bot/exts/moderation/infraction/management.py
Comment (mbaruh) [reply bot/exts/moderation/infraction/management.py:300]: I checked the categories, the ID's are correct.
Added the "closes" part

> I feel like we could make the first predicate simpler

Unless there's a matter of efficiency, I think the current line makes it pretty clear what it does.

**Review by Numerlor — APPROVED** [review]

**Review by lemonsaurus — APPROVED** [review]
This looks good, and is a nice solution. I also really like how thorough the PR description is. Nice work!

**Pull request merged**
